README.md

```diff
@@ -1,21 +1,43 @@
-bulletproof
+Bulletproof
 ===========
 
 [![Latest Stable Version](https://img.shields.io/packagist/v/mf/bulletproof.svg)](https://packagist.org/packages/mf/bulletproof)
 [![Build Status](https://travis-ci.org/mf/bulletproof.svg?branch=master)](https://travis-ci.org/mf/bulletproof)
 [![Coverage Status](https://coveralls.io/repos/github/mf/bulletproof/badge.svg?branch=master)](https://coveralls.io/github/mf/bulletproof?branch=master)
 
-{DESCRIPTION}
+The union of [PHPStan](https://github.com/phpstan/phpstan) type analysis (_type inference_) and property based testing by [Eris](https://github.com/giorgiosironi/eris).
 
-## Installation
+The idea is that [PHPStan](https://github.com/phpstan/phpstan) will deduce types of parameters and return type of the given method.
+Then `Bulletproof` (_currently `AbstractTestCase`_) lib will create generators from [Eris](https://github.com/giorgiosironi/eris) to generate the input for property based test.
+
+## Disclaimer
+@Ocramius said:
+> It would be cool to have a library which do that!
+
+So here it is ...
+
+Well I hope I (_or we 😉_) will implement this library in the future in the way we discuss it with @ocramius, but for now it is basically only the **idea**... 
+because [PHPStan](https://github.com/phpstan/phpstan) is no meant to be used like this now... so it could take some time...
+
+## The future
 
+I might (_and probably will_) do some implementation here, some more playing and prototyping of what it _would/could/should_ do in the future,
+but it will still be just a _naive and dummy_ implementation for now. 
+
+If you have some idea of your own feel free to open an issue or send PR and we can discuss it. 
```

Comment:
thank you for sharing some vision

Reply:
You are welcome :+1: 
Before I willl continue with this I'd like to see and try more with QuickCheck.. 

So I thought some info about this lib should be here for others..